### ISSUE TITLE
OKD 3.10 replace master fails using scaleup.yml playbook due to missing aggregator and front-proxy files

### ISSUE THREAD

**Issue by skynardo** [openshift/openshift-ansible#10657]
#### Description

After upgrading from Origin 3.9 to OKD 3.10 using openshift-ansible-3.10.51-1.git.0.44a646c.el7, we terminated our 2nd master (ec2 instance on AWS), rebuilt a new ec2 instance, then ran /usr/share/ansible/openshift-ansible/playbooks/openshift-master/scaleup.yml to replace the 2nd master. The playbook fails waiting to start master-api. Pod logs for controller and api on the new master indicate missing cert and key files.

##### Version

```
ansible 2.6.5
penshift-ansible-3.10.51-1.git.0.44a646c.el7.noarch
```

##### Steps To Reproduce
1. upgrade from Origin 3.9 to OKD 3.10 using openshift-ansible-3.10.51-1.git.0.44a646c.el7
2. terminate 2nd master (ec2 instance on AWS)
3. rebuild new ec2 instance to replace 2nd master
4. run: /usr/share/ansible/openshift-ansible/playbooks/openshift-master/scaleup.yml
5. fails with "waiting for control plane pods to become ready"

##### Expected Results
playbook to finish without errors and 2nd master again functional part of cluster
all pods in a running state as shown below:

```
oc adm manage-node ip-172-23-21-30.ec2.internal --list-pods

Listing matched pods on node: ip-172-23-21-30.ec2.internal

NAMESPACE                           NAME                                              READY     STATUS    RESTARTS   AGE
kube-service-catalog                apiserver-x8494                                   1/1       Running   1          3h
kube-service-catalog                controller-manager-2vxf6                          1/1       Running   1          3h
kube-system                         master-api-ip-172-23-21-30.ec2.internal           1/1       Running   24         3h
kube-system                         master-controllers-ip-172-23-21-30.ec2.internal   1/1       Running   10         3h
logging                             logging-fluentd-zl5gw                             1/1       Running   1          3h
openshift-node                      sync-j868d                                        1/1       Running   0          22m
openshift-sdn                       ovs-mz95f                                         1/1       Running   1          3h
openshift-sdn                       sdn-schdm                                         1/1       Running   1          3h
openshift-template-service-broker   apiserver-7j9rw                                   1/1       Running   1          3h
```

##### Observed Results
oc get pods on kube-system for the new master show api and controller pods in CrashLoopBackoffxample 

```
oc get pods
NAME                                              READY     STATUS             RESTARTS   AGE
master-api-ip-172-21-20-30.ec2.internal           1/1       Running            2          1d
master-api-ip-172-21-21-30.ec2.internal           1/1       Running            2          1d
master-api-ip-172-21-22-30.ec2.internal           0/1       CrashLoopBackOff   2          25s
master-controllers-ip-172-21-20-30.ec2.internal   1/1       Running            3          1d
master-controllers-ip-172-21-21-30.ec2.internal   1/1       Running            2          1d
master-controllers-ip-172-21-22-30.ec2.internal   0/1       CrashLoopBackOff   2          25s
```

For long output or logs, consider using a [gist](https://gist.github.com/)

##### Additional Information

We have found that the following files are missing from /etc/origin/master on the new master at the time that the scaleup playbook fails.
```
-rw-r--r--. 1 root root   1679 Nov  9 16:07 aggregator-front-proxy.key
-rw-r--r--. 1 root root   1090 Nov  9 16:07 aggregator-front-proxy.crt
-rw-r--r--. 1 root root   7982 Nov  9 17:15 aggregator-front-proxy.kubeconfig
-rw-r--r--. 1 root root   1070 Nov  9 17:19 front-proxy-ca.crt
-rw-r--r--. 1 root root   1679 Nov  9 17:19 front-proxy-ca.key
```
If we copy the above files from master0, then re-run the scaleup.yml playbook again, the playbook completes successfully and the master is again functional.

 Your operating system and version
CentOS Linux release 7.5.1804 (Core)

```
EXTRA INFORMATION GOES HERE
```

**Comment by vrutkovs** [comment]
What's in the logs of `master-api-ip-172-21-22-30.ec2.internal` and `master-controllers-ip-172-21-22-30.ec2.internal` containers?

**Comment by skynardo** [comment]
I apologize I did not capture the logs, I can recreate the issue if need be. Initially after the playbook fails, the logs indicate that they cannot find /etc/origin/master/aggregator-front-proxy.key and .crt files.  After I copied them from master[0] then issuing a master-restart on the api and controllers it complained about the front-proxy-ca.key and crt files being missing. After I copied those, I kicked off the scaleup.yml again and it ran to completion and the replaced master was again functional. If you'd like me to recreate and attach the logs, let me know.

**Comment by skynardo** [comment]
Here are the first errors received when the replacement master tries to start master-api pod,{"log":"Invalid MasterConfig /etc/origin/master/master-config.yaml\n","stream":"stderr","time":"2018-11-19T20:50:37.438392459Z"}
{"log":"  aggregatorConfig.proxyClientInfo.certFile: Invalid value: \"/etc/origin/master/aggregator-front-proxy.crt\": could not read file: stat /etc/origin/master/aggregator-front-proxy.crt: no such file or directory\n","stream":"stderr","time":"2018-11-19T20:50:37.438411308Z"}
{"log":"  aggregatorConfig.proxyClientInfo.keyFile: Invalid value: \"/etc/origin/master/aggregator-front-proxy.key\": could not read file: stat /etc/origin/master/aggregator-front-proxy.key: no such file or directory\n","stream":"stderr","time":"2018-11-19T20:50:37.438421726Z"}

After copying these 2 files from master0 to the replacement master1, it then complains about front-proxy.crt and front-proxy.key files missing.

**Comment by skynardo** [comment]
After further testing, it seems that replacing a previous master using the scaleup.yaml with the "workaround" described above does not actually work. We have found that after following the process described above, when trying to login to the cluster and hitting one of the other 2 original masters, the login fails with a Error from server (InternalError): Internal error occurred: unexpected response: 
When checking the master-api log file for the master that attempted to handle the login we see the error below.
400dccbd32c9fadb7a7f54553af07d7328/api/2.log:{"log":"E1120 21:06:37.470681       1 osinserver.go:107] internal error: urls don't validate: https://master.apps.shared.openshift.local:443/oauth/token/implicit / https://master.apps.shared.openshift.local/oauth/token/implicit\n","stream":"stderr","time":"2018-11-20T21:06:37.470850255Z"}

We also tried the latest 3.10 ansible playbooks openshift-ansible-3.10.68-1.git.0.f908cf5.el7 with the same results.  At this point we cannot use scaleup.yaml to replace a failed master.
